packages/snaps-cli/src/cmds/build/index.ts

```diff
@@ -16,8 +17,13 @@ export = {
       .option('src', builders.src)
       .option('stripComments', builders.stripComments)
       .option('transpilationMode', builders.transpilationMode)
+      .option('transpiledDeps', builders.transpiledDeps)
       .option('writeManifest', builders.writeManifest)
-      .implies('writeManifest', 'manifest');
+      .implies('writeManifest', 'manifest')
+      .implies('transpiledDeps', 'transpilationMode')
+      .middleware((argv) => {
+        processDependencies(argv as any);
+      });
```

Comment:
We should throw an error in the middleware if `argv.transpilationMode` is anything other than `localAndDeps` here, as a form of additional validation. The error message should state that the transpilation mode must be `localAndDeps`, and should ideally follow the format of the other validation errors thrown by yargs.

Reply:
Hmm, if we do that then we would be throwing an error on the other modes. I was thinking we can do something like `if (argv.transpilationMode === 'localAndDeps') processDependencies(argv as any)` and avoid throwing an error?

Reply:
The principle guiding my suggestion here is "fail fast" or "fail early". On the one hand, who cares if `transpiledDeps` doesn't do anything if the `transpilationMode` is something other than `localAndDeps`? It doesn't strictly hurt anyone or create an unsafe situation. However, it creates ambiguity, which is not a desirable property of any interface. In effect, we'd put the user in a situation where they're guessing how their input affects the behavior of our software. They'll have to ask themselves questions like "Are my dependencies transpiled anyway?", "Can I just transpile dependencies, but not my own source code?" etc. We don't want our users to guess what's happening; we want our interfaces to be unambiguous, and our applications to behave in a transparent manner.

In this light, throwing an error is not something to "avoid", but a way to communicate that the user's inputs violate the assumptions of our application. This is a good thing!

Reply:
I suppose I can just skip the middleware if `argv.transpiledDeps` doesn't exist and throw (log?) the error if it does and is anything other than `localAndDeps`.

Reply:
> the middleware is being hit regardless of if the `transpilationMode` flag is set or not.

Ah! Of course. The `.implies()` call will cause yargs to error if `transpiledDeps` is set _without_ `transpilationMode` being set, but the middleware will always be called. I forgot about that, my bad.

If `transpilationMode` is not set, the middleware should just no-op. If `transpilationMode` and `transpiledDeps` are set, and `transpilationMode` is **not** `localAndDeps`, we should exit with an error. Throwing should get the job done for us.

Reply:
Sounds good, will fix up ASAP.